.github/workflows/pytest.yml

```diff
@@ -0,0 +1,157 @@
+        required: true
+        type: string
+      config-fp:
+        required: false
+        type: string
+      config-pretty:
+        required: false
+        type: string
+        default: "[no config]"
+      python-version:
+        required: true
+        type: string
+      upstream-details:
+        required: false
+        type: string
+      pmlb-cache:
+        required: false
+        type: boolean
+        default: false
+      amalgam-plat-arch:
+        description: The platform/arch string to match when downloading Amalgam builds
+        required: false
+        type: string
+        default: "linux-amd64"
+      optional-dependencies:
+        description: An optional dependency installation specification for use with a pyproject.toml.
+        type: string
+        required: false
+        default: "[dev]"
+      no-self-install:
+        description: Set to 'true' if the current repository is not an installable Python package
+        type: boolean
+        required: false
+        default: false
+
+jobs:
+
+  test:
+    name: Pytest (${{ inputs.python-version }}, ${{inputs.platform-pretty}}, ${{ inputs.config-pretty }})
+    runs-on: ${{ inputs.platform }}
+    steps:
+
+      - uses: actions/checkout@v4
+
+      - name: Collect Workflow Telemetry
+        uses: catchpoint/workflow-telemetry-action@v1
+        with:
+          comment_on_pr: off
+          proc_trace_chart_show: off
+          proc_trace_table_show: off
+
+      - name: Pull GitHub cache
+        if: inputs.pmlb-cache == true
+        id: cache
+        uses: actions/cache@v3
+        env:
+          cache-name: dataset-cache-${{ inputs.platform-pretty }}
+        with:
+          path: ~/dataset_cache
+          key: dataset-cache-${{ inputs.platform-pretty }}
+      
+      - name: Setup PMLB Cache
+        if: inputs.pmlb-cache == true
+        run: |
+          ls -l ~/dataset_cache
+          mv ~/dataset_cache/dataset_cache.tar.gz dataset_cache.tar.gz && rm -rf dataset-cache
+          tar xzvf dataset_cache.tar.gz
+          ls .dataset_cache
+          cp -r .dataset_cache ~/.dataset_cache
+
+      - name: Download dependencies
+        env:
+          GH_TOKEN: ${{ secrets.HOWSOAI_WORKFLOW_AUTOMATION_CLASSIC_TOKEN }}
+        run: |
+          while read -r repo; do
+            echo "Getting details for $repo"
+            run_type=$(echo "${{ inputs.upstream-details }}" | jq --arg repo "$repo" '".$repo"."run-type"')
+            run_id=$(echo "${{ inputs.upstream-details }}" | jq --arg repo "$repo" '".$repo"."run-id"')
+            # Amalgam and howso-engine downloads need to be placed in their embedded location
+            if [[ "$repo" == "amalgam" ]]; then
+              gh $run_type download -D amalgam/lib/$plat/amd64 -R "howsoai/$repo" -p "*${{ inputs.amalgam-plat-arch }}*" "$run_id"
+              # Extract binaries
+              cd amalgam/lib
+              find . -type d -name lib -exec sh -c 'mv {}/* "$(dirname {})"' \;
```

Comment:
Does this line work? I think it should be similar to hse where it looks for the tarball first and them untars it to the right place.

Reply:
Added a line to unzip it, turns out in amalgam-lang-py we've just been using the release downloader action for this (which has an `extract` option)